contrib/core-contract-tests/tests/pox-4/pox-4.prop.test.ts

```diff
@@ -0,0 +1,1527 @@
+import { Cl, ClarityType, isClarityType } from "@stacks/transactions";
+import { assert, describe, expect, it } from "vitest";
+import fc from "fast-check";
+
+// Contracts
+const POX_4 = "pox-4";
+// Methods
+const GET_POX_INFO = "get-pox-info";
+const GET_STACKER_INFO = "get-stacker-info";
+const REWARD_CYCLE_TO_BURN_HEIGHT = "reward-cycle-to-burn-height";
+const BURN_HEIGHT_TO_REWARD_CYCLE = "burn-height-to-reward-cycle";
+const CURRENT_POX_REWARD_CYCLE = "current-pox-reward-cycle";
+const CHECK_CALLER_ALLOWED = "check-caller-allowed";
+const GET_REWARD_SET_SIZE = "get-reward-set-size";
+const GET_REWARD_SET_POX_ADDRESS = "get-reward-set-pox-address";
+const GET_TOTAL_USTX_STACKED = "get-total-ustx-stacked";
+const CHECK_POX_ADDR_VERSION = "check-pox-addr-version";
+const CHECK_POX_LOCK_PERIOD = "check-pox-lock-period";
+const GET_STACKING_MINIMUM = "get-stacking-minimum";
+const CAN_STACK_STX = "can-stack-stx";
+const MINIMAL_CAN_STACK_STX = "minimal-can-stack-stx";
+const GET_CHECK_DELEGATION = "get-check-delegation";
+const GET_DELEGATION_INFO = "get-delegation-info";
+const GET_ALLOWANCE_CONTRACT_CALLERS = "get-allowance-contract-callers";
+const ALLOW_CONTRACT_CALLER = "allow-contract-caller";
+// Contract Consts
+const TESTNET_STACKING_THRESHOLD_25 = 8000;
+const TESTNET_REWARD_CYCLE_LENGTH = 1050;
+const TESTNET_PREPARE_CYCLE_LENGTH = 50;
+const INITIAL_TOTAL_LIQ_SUPPLY = 1_000_000_000_000_000;
+const MIN_AMOUNT_USTX = 125_000_000_000n;
+// Clarity Constraints
+const MAX_CLAR_UINT = 340282366920938463463374607431768211455n;
+// Error Codes
+const ERR_STACKING_INVALID_LOCK_PERIOD = 2;
+const ERR_STACKING_THRESHOLD_NOT_MET = 11;
+const ERR_STACKING_INVALID_POX_ADDRESS = 13;
+const ERR_STACKING_INVALID_AMOUNT = 18;
```

Comment:
It's nice that all those are grouped (e.g. consts, error codes, etc.) and how about also sorting them alphabetically?

Reply:
I'll do it🫡